### ISSUE TITLE
Cannot switch models if haven't connected to the controller yet

### ISSUE THREAD

**Issue by bac** [juju/juju-gui#1544]
Launched GUI via charm.
Visited profile page: `https://localhost:8443/services/machine/profile/:flags:/blues/`

One model is shown. Clicked on it, got 'Manage' button. Clicked on it and the canvas briefly popped up but view remained on profile page. Opened console and saw:
```
switching to new socket URL: wss://localhost:8443/ws/api/10.157.213.248/17070/
Uncaught TypeError: Cannot set property 'onclose' of undefined
```

**Comment by hatched** [comment]
That socket url is invalid, I think the real issue here is how did that socket url get generated.

**Comment by bac** [comment]
Yes, I still see it using the same environment.
![screen shot 2016-04-12 at 11 40 20](https://cloud.githubusercontent.com/assets/617845/14466101/65e7c3ea-00a3-11e6-8ae6-03d3d60afe11.png)

**Comment by bac** [comment]
```
app.get('socket_url')
"wss://localhost:8443/ws/api/54.161.78.39/17070/b0e9e51f-541d-4490-892d-057f8f1866f7"
app.env.get('socket_url')
"wss://localhost:8443/ws/api/10.157.213.248/17070/"
```

**Comment by hatched** [comment]
If you load the GUI with `sandbox` off and `gisf` true the GUI doesn't connect to the controller to fetch things like the supported facades. 

This means that if you go to your profile and try and switch to a model it will fail with an invalid `socket_url` caused by the return value here https://github.com/juju/juju-gui/blob/develop/jujugui/static/gui/src/app/components/user-profile/user-profile.js#L103 which causes the callback to be called with an empty uuid because the `findfacade` calls return null when there is no facade data available.

This isn't an issue when you go through the full deployment flow because we pass all of the necessary information to the request bypassing the facade check.

**Comment by hatched** [comment]
#1557 fixes part of this issue, it now lets you switch to a new model except that the onclose call in `switchEnv` now fails because there isn't an open websocket. This can probably be fixed with #1555.
